src/typing/constraint_js.mli

```diff
@@ -266,7 +266,16 @@ type block_entry = {
   def_loc: Spider_monkey_ast.Loc.t option;
   for_type: bool;
 }
-type block = block_entry SMap.t ref
+
+type block_scope =
+  | Hoist (* var, class, functions hoisted up to this point *)
```

Comment:
Because variant names quickly lose the (often useful) context of the name of the type they're a part of, it's probably going to be more clear in callsites if you name these variants of the form `| HoistedScope | BlockScope`.

The "scope" becomes redundant here at the definition, but it becomes incredibly useful at callsites